uiowa/uiowa
# Event feature infused with schema.org

Relates to https://github.com/uiowa/uiowa/issues/5810
Resolves https://github.com/uiowa/uiowa/issues/5879

# To Do

- [x] Remaining styling (sub events enclosed and spanning full width)
- [x] Add subEvent urls to json+ld (https://mysite.uiowa.edu/node/123)
- [x] Parent event field label
- [x] Finalize path
- [x] Check onIowa and Admissions for compatibility
- [x] Change schedule heading to serif bold

# Follow up

- [ ] Deal with 404 caused by path pattern /event/123/event-title ( /event/123 -> /node/123 -> /event/123/event-title)
  - #5910 
- [ ] Add Event>eventStatus field (e.g. Status) for "active", "cancelled", or "rescheduled" events, attach to eventStatus property from schema???
  - #5939
- [ ] Add Thing>eventAttendanceMode field (e.g. Attendance Mode) for "online", "onsite" or "mixed/blended"???
  - #5940
  - consider adding this in preparation of an accepted property decision, considering this would be good to have


# To Test

- `git checkout schema_event`
- `ddev drush @default.local sql-drop -y && ddev drush @default.local cr`
- `ddev composer install && ddev yarn install && ddev yarn workspace uids_base build`
- `ddev blt di --site=default`
- login, enable the event split and `cim` twice.
- For the default site you should be able to create a parent event and then separately create sub events that link to the parent. The parent event should show the sub events in chronological order (nearest first). There is also a Presenter's field that allows referencing multiple people. There should no longer be an event categories field/taxonomy on the website.
- If you were to sync oniowa.uiowa.edu and admissions.uiowa.edu locally, you should see no import errors. For admissions, you should still be able to assign multiple event categories (sidebar) filter event by category on /events and in the events list block on the homepage.
- Admissions should not see the "Part of" or "Presenters" fields on their event form.
- Inspecting the HTML source, you should see a secondary json+ld script tag that contains event information if those fields are filled out. You should see superEvent if parent event is filled out, subEvent if sub events exists and a performer property if presenters is filled out. Additionally, location should include both address and virtual link if provided.
- Copy/pasting the inner source of the schema markup, you should get valid markup on https://validator.schema.org/
- Linking directly to source for events still works as expect (title and card body linked to the same URL on events list block and sub events view, admissions /events and that internal/external links work)
